experimentation/tools/sorald/handled_rules.py

```diff
@@ -0,0 +1,222 @@
+import argparse
+import dataclasses
+import pathlib
+import re
+import sys
+
+import jinja2
+
+from typing import Dict, List, Union
+
+from sorald._helpers import sonar_metadata, jsonkeys
+
+ENCODING = "utf8"
+
+TEMPLATE = r"""## Handled rules
+
+Sorald can currently repair violations of the following rules:
+
+* [Bug](#bug)
+{% for bug in bugs %}
+    * [{{ bug.title }}]({{ bug.link_to_detail }}) ([{{ bug.sonar_url_text }}]({{ bug.sonar_url }})) {% endfor %}
+
+* [Code Smell](#code-smell)
+{% for cs in code_smells %}
+    * [{{ cs.title }}]({{ cs.link_to_detail }}) ([{{ cs.sonar_url_text }}]({{ cs.sonar_url }})) {% endfor %}
+
+* [Vulnerability](#vulnerability)
+{% for vulnerability in vulnerabilities %}
+    * [{{ vulnerability.title }}]({{ vulnerability.link_to_detail }}) ([{{ vulnerability.sonar_url_text }}]({{ vulnerability.sonar_url }})) {% endfor %}
+
+### *Bug*
+{% for bug_content in bugs_detail %}
+#### {{ bug_content.title }} ([{{ bug_content.sonar_url_text }}]({{ bug_content.sonar_url }}))
+
+{{ bug_content.repair_description }}
+
+-----
+{% endfor %}
+### *Code Smell*
+{% for cs_content in code_smells_detail %}
+#### {{ cs_content.title }} ([{{ cs_content.sonar_url_text }}]({{ cs_content.sonar_url }}))
+
+{{ cs_content.repair_description }}
+
+-----
+{% endfor %}
+### *Vulnerability*
+{% for vulnerability_content in vulnerabilities_detail %}
+#### {{ vulnerability_content.title }} ([{{ vulnerability_content.sonar_url_text }}]({{ vulnerability_content.sonar_url }}))
+
+{{ vulnerability_content.repair_description }}
+
+-----
+{% endfor %}
+"""
+
+OUTPUT_ARG = "--output"
+
+PATH_TO_PROCESSOR_PACKAGE = pathlib.Path(__file__).absolute().parent.parent.parent.parent / "src/main/java/sorald/processor"
+
+
+@dataclasses.dataclass
+class RawSoraldProcessorInformation:
+    repair_description: str
+    rule_key: int
+
+
+@dataclasses.dataclass
+class ViolationList:
+    title: str
+    link_to_detail: str
+    sonar_url: str
+    sonar_url_text: str
+
+    def __lt__(self, other):
+        return self.title.lower() < other.title.lower()
+
+
+@dataclasses.dataclass
+class ViolationDetail:
+    title: str
+    sonar_url: str
+    sonar_url_text: str
+    repair_description: str
+
+    def __lt__(self, other):
+        return self.title.lower() < other.title.lower()
```

Comment:
> I don't see a reason for separating them, just create one dataclass with all of the information.

I separated them because the list with headings didn't have `repair_description` and the list of `repair_description` didn't require the link to itself so I decided to segregate them. If we keep them together, we can use the same list to render both the lists so that's an advantage for sure.